plugins/modules/s3_object.py

```diff
@@ -1217,15 +1225,17 @@ def main():
                 if key_check(module, s3, bucket, dirobj):
                     module.exit_json(msg="Bucket %s and key %s already exists." % (bucket, obj), changed=False)
                 else:
-                    # setting valid object acls for the create_dirkey function
-                    module.params['permission'] = object_acl
+                    if not acl_disabled:
+                        # setting valid object acls for the create_dirkey function
+                        module.params['permission'] = object_acl
                     create_dirkey(module, s3, bucket, dirobj, encrypt, expiry)
             else:
                 # only use valid bucket acls for the create_bucket function
                 module.params['permission'] = bucket_acl
                 create_bucket(module, s3, bucket, location)
-                # only use valid object acls for the create_dirkey function
-                module.params['permission'] = object_acl
+                if not acl_disabled:
+                    # only use valid object acls for the create_dirkey function
+                    module.params['permission'] = object_acl
```

Comment:
I don't think it's a good idea to modify `module.params` like this. The variable is supposed to be read-only. It becomes really hard to track the origin of the change.

Reply:
@goneri I have not changed the code flow much from what existed. I am not modifying the module.params['permission'], but setting the value based on a condition.

Reply:
Yes, I saw that the existing code is already doing that.